src/test/pmemblk_create/pmemblk_create.c

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2015, Intel Corporation
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in
+ *       the documentation and/or other materials provided with the
+ *       distribution.
+ *
+ *     * Neither the name of Intel Corporation nor the names of its
+ *       contributors may be used to endorse or promote products derived
+ *       from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/*
+ * pmemblk_create.c -- unit test for creating a block memory pool
+ *
+ * usage: pmemblk_create path bsize
+ */
+
+#include "unittest.h"
+
+#define	SIZEOF_NULLFILE_STR	(8)
+#define	SIZEOF_TESTFILE		(64 * 1024 * 1024)
+#define	CREATE_MODE			(0664)
+
+int
+main(int argc, char *argv[])
+{
+	START(argc, argv, "pmemblk_create");
+
+	if (argc != 3)
+		FATAL("usage: %s path bsize", argv[0]);
+
+	PMEMblkpool *handle;
+	const char *path = argv[1];
+	size_t bsize = strtoul(argv[2], NULL, 0);
+
+	if (memcmp(path, "NULLFILE", SIZEOF_NULLFILE_STR)) {
```

Comment:
Yes, I agree with the first comment, and that change has been made.  However, regarding the second comment, I respectfully disagree.  Using "argc" to determine whether or not a path has been provided is both ambiguous and error-prone.  For example, using this technique, how would one easily distinguish between the case where a path was not provided, or it was provided but the user simply forgot to include the last argument?  There's more than one way to skin a cat, and surely with enough code, we can make this work any way we like.  But, in my opinion, the use of a special string (or token, if you will) called NULLFILE is a simple, elegant, and effective way to convey the user's intention to not pass in a file.  And it avoids the additional complexity of having the program support a variable number of arguments.  I will change it if you feel strongly about it, but for now I will stand by this technique pending further discussion.